usr/src/uts/armv8/os/ddi_impl.c

```diff
@@ -1031,31 +1369,84 @@ get_boot_properties(void)
 			continue;
 		}
 		BOP_GETPROP(bootops, property_name, bop_staging_area);
+		flags = do_bsys_getproptype(bootops, property_name);
 
-		if (strcmp(property_name, "si-machine") == 0) {
-			(void) strncpy(utsname.machine, bop_staging_area, SYS_NMLN);
-			utsname.machine[SYS_NMLN - 1] = 0;
-		} else if (strcmp(property_name, "si-hw-provider") == 0) {
+		/*
+		 * special properties:
+		 * si-machine, si-hw-provider
+		 *	goes to kernel data structures.
+		 * bios-boot-device and stdout
+		 *	goes to hardware property list so it may show up
+		 *	in the prtconf -vp output. This is needed by
+		 *	Install/Upgrade. Once we fix install upgrade,
```

Comment:
Is this comment from i86pc?  We should probably find a better name than `bios-boot-device` if it's relevant to us, but I'm not sure how it could be right now.

Reply:
Yeah, the code is almost completely transposed from i86pc - with a handful of Arm adjustments.

I need to flesh out my `loader64.efi` work before I can tell if this would be relevant to us, but at first blush I don't think it would be.